scripts/segmentation/test.py

```diff
@@ -24,58 +77,120 @@ def test(args):
         transforms.ToTensor(),
         transforms.Normalize([.485, .456, .406], [.229, .224, .225]),
     ])
-    # dataset and dataloader
+    data_kwargs = {'transform': input_transform, 'base_size': args.base_size,
+                   'crop_size': args.crop_size}
+    # get dataset
     if args.eval:
         testset = get_segmentation_dataset(
-            args.dataset, split='val', mode='testval', transform=input_transform)
-        total_inter, total_union, total_correct, total_label = \
-            np.int64(0), np.int64(0), np.int64(0), np.int64(0)
+            args.dataset, split='val', mode=args.mode, **data_kwargs)
     else:
         testset = get_segmentation_dataset(
-            args.dataset, split='test', mode='test', transform=input_transform)
+            args.dataset, split='test', mode=args.mode, **data_kwargs)
+    size = len(testset)
+
+    # get dataloader
+    batchify_fn = ms_batchify_fn if args.mode == 'test' else None
     test_data = gluon.data.DataLoader(
-        testset, args.test_batch_size, shuffle=False, last_batch='keep',
-        batchify_fn=ms_batchify_fn, num_workers=args.workers)
-    # create network
-    if args.model_zoo is not None:
-        model = get_model(args.model_zoo, pretrained=True)
-    else:
-        model = get_segmentation_model(model=args.model, dataset=args.dataset, ctx=args.ctx,
-                                       backbone=args.backbone, norm_layer=args.norm_layer,
-                                       norm_kwargs=args.norm_kwargs, aux=args.aux,
-                                       base_size=args.base_size, crop_size=args.crop_size)
-        # load pretrained weight
-        assert args.resume is not None, '=> Please provide the checkpoint using --resume'
-        if os.path.isfile(args.resume):
-            model.load_parameters(args.resume, ctx=args.ctx)
-        else:
-            raise RuntimeError("=> no checkpoint found at '{}'" \
-                .format(args.resume))
+            testset, args.batch_size, batchify_fn=batchify_fn, last_batch='keep', shuffle=False, num_workers=args.workers)
+
     print(model)
-    evaluator = MultiEvalModel(model, testset.num_class, ctx_list=args.ctx)
```

Comment:
This would break the exisiting multi-size evaluation code. Could you change this script to other filename instead of overwriting `test.py`?

Reply:
Since quantized models are generated based on fixed `crop_size`, so I cannot use `MultiEvalModel` class here. Can we add a new function or a new script for INT8 inference? @xinyu-intel

Reply:
can use a separate function ‘def quantized_test’